docs/hub/agents.md

```diff
@@ -0,0 +1,112 @@
+# Agents on the Hub
+
+This page compiles all the libraries and tools Hugging Face offers for agentic workflows: huggingface.js mcp-client, Gradio MCP Server and smolagents.
+
+## huggingface.js mcp-client
+
+Huggingface.js offers an MCP client served with [Inference Providers](https://huggingface.co/docs/inference-providers/en/index) or local LLMs. Getting started with them is as simple as running `pnpm agent`. You can plug and play different models and providers by setting `PROVIDER` and `MODEL_ID` environment variables. 
+
+```bash
+export HF_TOKEN="hf_..."
+export MODEL_ID="Qwen/Qwen2.5-72B-Instruct"
+export PROVIDER="nebius"
+npx @huggingface/mcp-client
+```
+
+or, you can use any Local LLM (for example via lmstudio):
+
+```bash
+ENDPOINT_URL=http://localhost:1234/v1 \
+MODEL_ID=lmstudio-community/Qwen3-14B-GGUF \
+npx @huggingface/mcp-client
+```
+
+You can get more information about mcp-client [here](https://huggingface.co/docs/huggingface.js/en/mcp-client/README).
+
+
+## Gradio MCP Server / Tools
+
+You can build an MCP server in just a few lines of Python with Gradio. If you have an existing Gradio app or Space you'd like to use as an MCP server / tool, it's just a single-line change.
+
+To make a Gradio application an MCP server, simply pass in `mcp_server=True` when launching your demo like follows.
+
+```python
+# pip install gradio
+
+import gradio as gr
+
+def generate_image(prompt: str):
+   """
+   Generate an image based on a text prompt
+   
+   Args:
+       prompt: a text string describing the image to generate
+   """
+   pass
+
+demo = gr.Interface(
+    fn=generate_image,
+    inputs="text",
+    outputs="image",
+    title="Image Generator"
+)
+
+demo.launch(mcp_server=True)
+```
+
+The MCP server will be available at `http://your-server:port/gradio_api/mcp/sse` where your application is served. It will have a tool corresponding to each function in your Gradio app, with the tool description automatically generated from the docstrings of your functions.
+
+Lastly, add this to the settings of the MCP Client of your choice (e.g. Cursor).
+
+```json
+{
+  "mcpServers": {
+    "gradio": {
+      "url": "http://your-server:port/gradio_api/mcp/sse"
+    }
+  }
+}
+```
+
+This is very powerful because it lets the LLM use any Gradio application as a tool. You can find thousands of them on [Spaces](https://huggingface.co/spaces). 
```

Comment:
we can link some docs for people to read more: https://www.gradio.app/guides/building-mcp-server-with-gradio